src/corehost/cli/fxr/fx_ver.h

```diff
@@ -6,8 +6,7 @@
 
 #include "pal.h"
 
-// Note: This is not SemVer (esp., in comparing pre-release part, fx_ver_t does not
-// compare multiple dot separated identifiers individually.) ex: 1.0.0-beta.2 vs. 1.0.0-beta.11
+// Note: This is intended to implement SemVer 2.0
 struct fx_ver_t
 {
     fx_ver_t();
```

Comment:
Can't add a comment to the .ctor which takes all the values. I think we should add an assert to it that validates that `m_pre` starts with a hyphen (if it's not empty) and that `m_build` starts with a `+` if it's not empty.

Reply:
> Can't add a comment to the .ctor which takes all the values.

Why?

Reply:
Sorry for the confusion - the "comment" is referring to me adding a PR comment - not a comment in the code.
Somehow GitHub doesn't let me add a comment to a place in the file which is not changed... maybe I'm doing it wrong.